### PR TITLE
hyprctl: add --quiet flag

### PR DESCRIPTION
**Pull request by diniamo**
#### Describe your PR, what does it fix/add?

Adds a `--quiet` flag to hyprctl that supresses its output.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

Might want to use `setstate(failbit)` instead of `rdbuf(nullptr)`, I'm not sure.

#### Is it ready for merging, or does it need work?

It's ready.

### CODE REVIEW

File: hyprctl/main.cpp [hunk 1]
```diff
@@ -341,6 +344,10 @@ int main(int argc, char** argv) {
         fullRequest += ARGS[i] + " ";
     }
 
+    if (quiet) {
+        std::cout.rdbuf(nullptr);
```
Comment (vaxerski) [inline hyprctl/main.cpp:348]: I'd rather wrap std::cout in something like `log` that does `if (quiet) return;`

**Comment by diniamo** [comment]
How is this?

Note:
- I left normal cout for usage prints, should those also be included?
- Removed the endl print at the end, since I highly doubt it's necessary, and is also annoying

File: hyprctl/main.cpp [hunk 2]
```diff
@@ -30,6 +30,7 @@
 #define PAD
 
 std::string instanceSignature;
+bool        quiet;
```
Comment (vaxerski) [inline hyprctl/main.cpp:33]: please initialize primitives

File: hyprctl/main.cpp [hunk 3]
```diff
@@ -39,6 +40,11 @@ struct SInstanceData {
     bool        valid = true;
 };
 
+void log(std::string str) {
+    if (!quiet)
```
Comment (vaxerski) [inline hyprctl/main.cpp:44]: if (quiet)
   return;

File: hyprctl/main.cpp [hunk 4]
```diff
@@ -419,6 +427,5 @@ int main(int argc, char** argv) {
         exitStatus = request(fullRequest);
     }
 
-    std::cout << std::endl;
```
Comment (vaxerski) [inline hyprctl/main.cpp:422]: this is useful as it flushes stdout, though you can also just flush

File: hyprctl/main.cpp (hunk 3 above)
Comment (diniamo) [reply hyprctl/main.cpp:44]: Why?

File: hyprctl/main.cpp (hunk 4 above)
Comment (diniamo) [reply hyprctl/main.cpp:422]: Why is flushing useful?

File: hyprctl/main.cpp (hunk 3 above)
Comment (vaxerski) [reply hyprctl/main.cpp:44]: always prefer guards, they are cleaner.

File: hyprctl/main.cpp (hunk 4 above)
Comment (vaxerski) [reply hyprctl/main.cpp:422]: to avoid half-written logs

**Comment by diniamo** [comment]
That should be all.

**Review by vaxerski — APPROVED** [review]
lgtm thanks!

**Pull request merged**